templates/generate_docs.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+# mypy: ignore-errors
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import os
+from pathlib import Path
+
+import yaml
+from jinja2 import Environment, FileSystemLoader
+
```

Comment:
Might be nice to be able to run this script from top-level. Pulling inspiration from Triton's documentation generation script, we could add:

```python3
# Global constants
PA_ABSLT_PATH = os.environ.get("PA_ABSLT_PATH", os.getcwd())
PA_TEMPLATES_ABSLT_PATH = os.path.join(PA_ABSLT_PATH, "templates")

# Change working directory to perf_analyzer/templates.
os.chdir(PA_TEMPLATES_ABSLT_PATH)
.
.
# Render logic
.
.
# Change working directory to perf_analyzer.
os.chdir(PA_ABSLT_PATH)
```

Reply:
Added this. Now we can run the script from the root directory.